### PR TITLE
[ch37] Upgrade Authorize Net to support sha-512 hash

### PR DESCRIPTION
**Pull request by jvkoh**
Also upgrades the `nokogiri` bundle to fix a security issue.

### CODE REVIEW

File: lib/authorize_net/api.rb [hunk 1]
```diff
@@ -280,21 +280,17 @@ def handleResponse(raw_response)
   #
   # @throws AuthorizeNet::Exception
   # =============================================
-  def validate_hash(response_xml, amount, options={})
-    if @md5_hash.nil?
+  def validate_hash(response_xml, amount)
+    if @signature_key.nil?
```
Comment (jvkoh) [inline lib/authorize_net/api.rb:284]: This seems fine?  Although I can also imagine taking this out and enforcing hash validation

File: authorize_net.gemspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Gem::Specification.new do |spec|
   spec.name = 'authorize_net'
-  spec.version = '0.1.2'
+  spec.version = '1.0.0'
   spec.date = '2016-06-04'
```
Comment (cassianrx) [inline authorize_net.gemspec:4]: @jvkoh should this date be updated?

Comment (jvkoh) [reply authorize_net.gemspec:4]: Looks like that probably shouldn't exist at all actually.

Date is determined automatically when you package a gem, and the only thing I can find about this variable says not to set it: https://ruby-doc.org/stdlib-2.3.0/libdoc/rubygems/rdoc/Gem/Specification.html

Comment (cassianrx) [reply authorize_net.gemspec:4]: Sounds like we should remove it then 😛

**Pull request merged**